File: C++/string/Integer to Roman.cpp
lass Solution {
public:
    string intToRoman(int num) {
        vector<string>Symbol = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        vector<int>Numbers = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
        string Result = "";
        for(int i=0;i<Symbol.size() && num>0;i++)
        {
            while(Numbers[i]<=num)
            {
                num -= Numbers[i];
                Result += Symbol[i];
            }
        }
        
        return Result;
    }
};

File: Meta/Practice/Python/Integer to Roman.py
'''
Example 1:

Input: num = 3
Output: "III"
Explanation: 3 is represented as 3 ones.
Example 2:

Input: num = 58
Output: "LVIII"
Explanation: L = 50, V = 5, III = 3.
'''
'''
Constraints:-
1 <= num <= 3999
'''
'''
Approach:-
Greedy
Tc:- O(1) the num is 3999 which is still constant
Sc:- O(1) map is constant space
'''
class Solution:
    def intToRoman(self, num: int) -> str:
        # Creating Dictionary for Lookup
        num_map = {
            1: "I",
            5: "V",    4: "IV",
            10: "X",   9: "IX",
            50: "L",   40: "XL",
            100: "C",  90: "XC",
            500: "D",  400: "CD",
            1000: "M", 900: "CM",
        }
        
        # Result Variable
        r = ''
        
        
        for n in [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]: # O(1) since this is a constant list
            # If n in list then add the roman value to result variable
            while n <= num:
                r += num_map[n]
                num-=n
        return r
        
